homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,353 @@
+            step_id=STEP_HUB_V1, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_hubv2(self, user_input=None):
+        """Set up the Hub v1 modem type."""
+        errors = {}
+        if user_input is not None:
+            user_input[CONF_HUB_VERSION] = 2
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            user_input.pop(CONF_HUB_VERSION)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_hub_v2_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_HUB_V2, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_import(self, import_info):
+        """Import a new connection as a config entry."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        if not await _async_connect(**import_info):
+            return self.async_abort(reason="cannot_connect")
+        return self.async_create_entry(title="", data=import_info)
+
+
+class InsteonOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle an Insteon options flow."""
+
+    def __init__(self, config_entry):
+        """Init the InsteonOptionsFlowHandler class."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Init the config flow."""
+        errors = {}
+        if user_input is not None:
+            change_hub_config = user_input.get(STEP_CHANGE_HUB_CONFIG, False)
+            device_override = user_input.get(STEP_ADD_OVERRIDE, False)
+            x10_device = user_input.get(STEP_ADD_X10, False)
+            remove_override = user_input.get(STEP_REMOVE_OVERRIDE, False)
+            remove_x10 = user_input.get(STEP_REMOVE_X10, False)
+            if _only_one_selected(
+                change_hub_config,
+                device_override,
+                x10_device,
+                remove_override,
+                remove_x10,
+            ):
+                if change_hub_config:
+                    return await self.async_step_change_hub_config()
+                if device_override:
+                    return await self.async_step_add_override()
+                if x10_device:
+                    return await self.async_step_add_x10()
+                if remove_override:
+                    return await self.async_step_remove_override()
+                if remove_x10:
+                    return await self.async_step_remove_x10()
+            errors["base"] = "select_single"
+
+        data_schema = {
+            vol.Optional(STEP_ADD_OVERRIDE): bool,
+            vol.Optional(STEP_ADD_X10): bool,
+        }
+        if self.config_entry.data.get(CONF_HOST):
+            data_schema[vol.Optional(STEP_CHANGE_HUB_CONFIG)] = bool
+
+        options = {**self.config_entry.options}
```

Comment:
```suggestion
        options = self.config_entry.options
```
Do you need a copy here?